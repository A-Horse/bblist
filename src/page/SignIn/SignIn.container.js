import { connect } from "react-redux";
import SignIn from "./SignIn";
import { withRouter } from "react-router-dom";
import { bindActionCreators } from "redux";
import Actions, { makeActionRequestCollection } from "../../actions/actions";

const mapDispatchToProps = dispatch => {
  return {
    actions: bindActionCreators(
      makeActionRequestCollection([Actions.LOGIN]),
      dispatch
    )
  };
};

const mapStateToProps = state => {
  return {
    signInErrorMessages: state.auth.get("signInErrorMessage"),
    signInAuthenticated: state.auth.get("signInAuthenticated")
  };
};

export const SignInContainer = withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(SignIn)
);

export default SignInContainer;
