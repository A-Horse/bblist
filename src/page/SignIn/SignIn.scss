@import "../../style/variable";
@import "../../style/mixin";

.signin-page {
  display: flex;
  justify-content: center;
  width: 100%;
  box-sizing: border-box;

  .signin-main {
    margin-top: 80px;
    width: 450px;
    box-shadow: 1px 2px 10px #e1e1e1;
    padding: 23px 40px;
    border-radius: 8px;
  }
}
