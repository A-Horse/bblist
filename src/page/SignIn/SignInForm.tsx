import { Form, Icon, Input, Button } from "antd";
import React, { Component, FormEvent } from "react";
import { WrappedFormUtils } from "antd/lib/form/Form";

interface Props {
  form: WrappedFormUtils;
  onSubmit: (values: any) => void;
}

export interface SignInData {
  email: string;
  password: string;
}

class SignInForm extends Component<Props> {
  handleSubmit = (e: FormEvent) => {
    e.preventDefault();
    this.props.form.validateFields((err: Error, values: SignInData) => {
      if (!err) {
        this.props.onSubmit(values);
      }
    });
  };

  render() {
    const { getFieldDecorator } = this.props.form;
    return (
      <Form onSubmit={this.handleSubmit} className="login-form">
        <Form.Item>
          {getFieldDecorator("email", {
            rules: [{ required: true, message: "Please input your email!" }]
          })(
            <Input
              size="large"
              type="email"
              prefix={<Icon type="user" style={{ color: "rgba(0,0,0,.25)" }} />}
              placeholder="Email"
            />
          )}
        </Form.Item>
        <Form.Item>
          {getFieldDecorator("password", {
            rules: [{ required: true, message: "Please input your Password!" }]
          })(
            <Input
              size="large"
              prefix={<Icon type="lock" style={{ color: "rgba(0,0,0,.25)" }} />}
              type="password"
              placeholder="Password"
            />
          )}
        </Form.Item>
        <Form.Item>
          <Button
            type="primary"
            htmlType="submit"
            className="login-form-button"
          >
            Log in
          </Button>
        </Form.Item>
      </Form>
    );
  }
}

export const WrappedSignInForm = Form.create()(SignInForm);
