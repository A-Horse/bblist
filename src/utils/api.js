import { API_PREFIX, LOCAL_BACKUP_URL } from '../constants';
import { Storage } from 'services/storage';

export function makeApiUrl(path) {
  if (Storage.get(LOCAL_BACKUP_URL)) {
    return `${Storage.get(LOCAL_BACKUP_URL)}${path}`;
  }
  return `${API_PREFIX}${path}`;
}
