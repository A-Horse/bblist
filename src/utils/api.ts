export function makeApiUrl(path) {
  return `/api${path}`;
}
