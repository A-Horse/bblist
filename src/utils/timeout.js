import { Promise } from "es6-promise";

export function timeout(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}
