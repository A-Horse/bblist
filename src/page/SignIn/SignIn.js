//
import React, { Component } from 'react';
import { validateFormValue } from '../../services/validate-strategy';
import { Button } from '../../components/widget/Button/Button';
import { Input } from '../../components/widget/Input/Input';
import { LogoBan } from '../../components/commons/LogoBan/LogoBan';
import { ErrorMsg } from '../../components/ErrorMsg/ErrorMsg';
import { PageContainer } from '../../components/widget/PageContainer';
import { Link } from 'react-router-dom';
import { updateTitle } from '../../services/title';

import * as R from 'ramda';

import './SignIn.scss';

class SignIn extends Component {
  state = {
    errorMessages: {},
    email: '',
    password: ''
  };

  componentDidMount() {
    updateTitle('Sign In');
  }

  componentWillReceiveProps(newProps) {
    if (newProps.signInAuthenticated) {
      this.props.actions.LOGIN_FINISH();
      this.props.history.push('/home');
    }
  }

  render() {
    return (
      <PageContainer className="signin-page">
        <div className="signin-main">
          <LogoBan />
          <form className="signin-form" onSubmit={this.login}>
            <div>
              <Input
                type="text"
                name="octopus-email"
                required
                placeholder="Email"
                onChange={value => this.setState({ email: value })}
              />
            </div>

            <div>
              <Input
                type="password"
                name="octopus-password"
                onChange={value => this.setState({ password: value })}
                required
                placeholder="Password"
              />
            </div>

            <ErrorMsg
              messages={R.compose(
                R.concat([this.props.signInErrorMessages]),
                R.values
              )(this.state.errorMessages)}
            />
            <Button className="signin-button" size="large" type="submit" styleType="primary">
              Login
            </Button>
          </form>

          <div className="signup-tip">
            Do not have an account yet?
            <Link className="signup-link" to="/signup">
              Sign up
            </Link>
          </div>
        </div>
      </PageContainer>
    );
  }

  login = event => {
    event.preventDefault();
    const loginInfo = {
      email: this.state.email.trim(),
      password: this.state.password.trim()
    };

    const errorMessages = validateFormValue(loginInfo, {
      email: ['email'],
      password: ['max@100#Password Up to 100 characters', 'min@6#min 6']
    });

    this.setState({ errorMessages: errorMessages });
    if (Object.keys(errorMessages).length) {
      return;
    }
    this.props.actions.LOGIN_REQUEST(loginInfo);
  };
}

export default SignIn;
