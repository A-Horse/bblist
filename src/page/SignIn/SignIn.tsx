import React, { Component } from 'react';

import { PageContainer } from '../../components/widget/PageContainer';
import { Link } from 'react-router-dom';
import { updateTitle } from '../../services/title';
import { WrappedSignInForm } from './SignInForm';
import { Logo } from '../../components/commons/Logo/Logo';


import './SignIn.scss';
import { TextLogo } from '../../components/commons/TextLogo';

interface Props {
  actions: any;
  history: any;
  signInErrorMessages: any;
}

class SignIn extends Component<Props> {
  state = {
    errorMessages: {},
    email: '',
    password: ''
  };

  componentDidMount() {
    updateTitle('Sign In');
  }

  componentWillReceiveProps(newProps: any) {
    if (newProps.signInAuthenticated) {
      this.props.actions.LOGIN_FINISH();
      this.props.history.push('/home');
    }
  }

  render() {
    return (
      <div className="signin-page">
        <div className="signin-main">
          <Logo style={{
            width: 45
          }} />
          
          <TextLogo />

          <WrappedSignInForm onSubmit={this.login} />

          <div className="signup-tip">
            Do not have an account yet?
            <Link className="signup-link" to="/signup">
              Sign up
            </Link>
          </div>
        </div>
      </div>
    );
  }

  login = (signInData: any) => {
    this.props.actions.LOGIN_REQUEST(signInData);
  };
}

export default SignIn;
