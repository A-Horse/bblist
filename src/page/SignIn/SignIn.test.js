import React from 'react';
import SignIn from './SignIn';
import { timeout } from 'utils/timeout';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import { fromJS } from 'immutable';
import { mount } from 'enzyme';

test('SignIn', () => {
  const actions = {};
  const history = { push: jest.fn() };
  const signInErrorMessages = null;

  const tree = renderer.create(
    <MemoryRouter>
      <SignIn actions={actions} history={history} signInErrorMessages={signInErrorMessages} />
    </MemoryRouter>
  );
  expect(tree).toMatchSnapshot();
});

test('SignIn signin', () => {
  const actions = { LOGIN_REQUEST: jest.fn() };
  const history = { push: jest.fn() };
  const signInErrorMessages = null;

  const wrapper = mount(
    <MemoryRouter>
      <SignIn actions={actions} history={history} signInErrorMessages={signInErrorMessages} />
    </MemoryRouter>
  );

  const emailInput = wrapper.find('input[name="octopus-email"]');
  const passwdInput = wrapper.find('input[name="octopus-password"]');

  emailInput.simulate('focus');
  emailInput.node.value = 'octopus@octopus.com';
  emailInput.simulate('change', { target: { value: 'octopus@octopus.com' } });

  passwdInput.simulate('focus');
  passwdInput.node.value = 'octopus';
  passwdInput.simulate('change', { target: { value: 'octopus' } });

  wrapper
    .find('.signin-form')
    .simulate('focus')
    .simulate('submit');

  expect(actions.LOGIN_REQUEST.mock.calls.length).toBe(1);
});
