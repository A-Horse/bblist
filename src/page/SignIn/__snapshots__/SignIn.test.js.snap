// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn 1`] = `
<div
  className="page-container signin-page"
>
  <div
    className="signin-main"
  >
    <div
      className="logoban"
      onClick={[Function]}
    >
      <img
        className="logo"
        src="/static/logo.png"
      />
      <img
        className="text-logo"
        src="/static/text-logo.svg"
      />
    </div>
    <form
      className="signin-form"
      onSubmit={[Function]}
    >
      <div>
        <input
          name="octopus-email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="text"
        />
      </div>
      <div>
        <input
          name="octopus-password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <noscript />
      <button
        className="button large primary border-default  signin-button"
        disabled={undefined}
        onClick={undefined}
        type="submit"
      >
        Login
      </button>
    </form>
    <div
      className="signup-tip"
    >
      Do not have an account yet?
      <a
        className="signup-link"
        href="/signup"
        onClick={[Function]}
      >
        Sign up
      </a>
    </div>
  </div>
</div>
`;
